src/optimizer/rule_query_rewrite.cpp

```diff
@@ -0,0 +1,191 @@
+
+  if (new_predicates.empty())
+    return;
+
+  std::vector<AnnotatedExpression> total_predicates;
+
+  for (auto &predicate : get_predicates) {
+    total_predicates.emplace_back(predicate);
+  }
+
+  for (auto &predicate : new_predicates) {
+    total_predicates.emplace_back(predicate);
+  }
+  
+  std::shared_ptr<OperatorExpression> output =
+      std::make_shared<OperatorExpression>(
+          LogicalGet::make(get->get_id, total_predicates,
+  			   get->table, get->table_alias, get->is_for_update));
+  
+  transformed.push_back(output);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+/// LogicalFilter
+TransitivePredicatesLogicalFilter::TransitivePredicatesLogicalFilter() {
+  type_ = RuleType::TRANSITIVE_PREDICATES_LOGICAL_FILTER;
+
+  std::shared_ptr<Pattern> child(std::make_shared<Pattern>(OpType::InnerJoin));
+  child->AddChild(std::make_shared<Pattern>(OpType::Leaf));
+  child->AddChild(std::make_shared<Pattern>(OpType::Leaf));
+
+  match_pattern = std::make_shared<Pattern>(OpType::LogicalFilter);
+
+  match_pattern->AddChild(child);
+}
+
+bool TransitivePredicatesLogicalFilter::Check(std::shared_ptr<OperatorExpression> input,
+					      OptimizeContext *context) const {
+  (void)input;
+  (void)context;
+  return true;
+}
+
+void TransitivePredicatesLogicalFilter::Transform(
+    std::shared_ptr<OperatorExpression> input,
+    std::vector<std::shared_ptr<OperatorExpression>> &transformed,
+    OptimizeContext *context) const {
+  LOG_TRACE("TransitivePredicatesLogicalFilter::Transform");
+
+  auto filter = input->Op().As<LogicalFilter>();
+  auto &filter_predicates = filter->predicates;
+
+  util::FillTransitiveTable(filter_predicates, context->transitive_table);
+  auto new_predicates = util::GenerateTransitivePredicates(filter_predicates, context->transitive_table);
+
+  if (new_predicates.empty())
+    return;
+
+  std::vector<AnnotatedExpression> total_predicates;
+
+  for (auto &predicate : filter_predicates) {
+    total_predicates.emplace_back(predicate);
+  }
+
+  for (auto &predicate : new_predicates) {
+    total_predicates.emplace_back(predicate);
+  }
+  
+  std::shared_ptr<OperatorExpression> output =
+      std::make_shared<OperatorExpression>(
+          LogicalFilter::make(total_predicates));
+
+  auto filter_children = input->Children();
+  for (auto child : filter_children) {
+    output->PushChild(child);
+  }
+
+  transformed.push_back(output);
+}
+
+//===--------------------------------------------------------------------===//
+// Simplify predicates rules
+//===--------------------------------------------------------------------===//
+
+///////////////////////////////////////////////////////////////////////////////
+/// LogicalFilter
+SimplifyPredicatesLogicalFilter::SimplifyPredicatesLogicalFilter() {
+  type_ = RuleType::SIMPLIFY_PREDICATES_LOGICAL_FILTER;
+
+  std::shared_ptr<Pattern> child(std::make_shared<Pattern>(OpType::InnerJoin));
```

Comment:
Same comment as in earlier LogicalFilter rule, is there a reason for this to be an InnerJoin and not a leaf?